SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,806 @@
+}
+
+- (void)sdl_handleTransportRegisterTimeout {
+    // if the state is still Connecting, go back to Configured state and retry immediately
+    if ([self.stateMachine isCurrentState:SDLSecondaryTransportStateConnecting]) {
+        SDLLogD(@"Retry secondary transport connection after registration timeout");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+    }
+}
+
+- (void)sdl_handleProtocolClosed {
+    if ([self sdl_isTransportOpened]) {
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateReconnecting];
+    }
+}
+
+- (void)sdl_handleAppBecomeActive {
+    if (([self.stateMachine isCurrentState:SDLSecondaryTransportStateConfigured])
+        && self.transportSelection == SDLTransportSelectionTCP && [self sdl_isTCPReady]) {
+        SDLLogD(@"Resuming TCP transport since the app becomes foreground");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConnecting];
+    }
+}
+
+- (void)sdl_handleAppResignedActive {
+    if ([self sdl_isTransportOpened] && self.transportSelection == SDLTransportSelectionTCP) {
+        SDLLogD(@"Disconnecting TCP transport since the app will go to background");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+    }
+}
+
+- (BOOL)sdl_isTransportOpened {
+    return [self.stateMachine isCurrentState:SDLSecondaryTransportStateConnecting] ||
+           [self.stateMachine isCurrentState:SDLSecondaryTransportStateRegistered];
+}
+
+
+#pragma mark - Starting / Stopping / Restarting services
+
+- (void)sdl_handleTransportUpdate:(BOOL)primaryAvailable secondaryAvailable:(BOOL)secondaryAvailable {
+    // update audio service
+    [self sdl_updateService:SDLServiceTypeAudio
+          allowedTransports:self.transportsForAudioService
+           primaryAvailable:primaryAvailable
+         secondaryAvailable:secondaryAvailable
+      transportUpdatedBlock:^(SDLProtocol * _Nullable oldProtocol, SDLProtocol * _Nullable newProtocol) {
+          [self.streamingProtocolListener onAudioServiceProtocolUpdated:oldProtocol to:newProtocol];
+      }];
+
+    // update video service
+    [self sdl_updateService:SDLServiceTypeVideo
+          allowedTransports:self.transportsForVideoService
+           primaryAvailable:primaryAvailable
+         secondaryAvailable:secondaryAvailable
+      transportUpdatedBlock:^(SDLProtocol * _Nullable oldProtocol, SDLProtocol * _Nullable newProtocol) {
+          [self.streamingProtocolListener onVideoServiceProtocolUpdated:oldProtocol to:newProtocol];
+      }];
+}
+
+- (void)sdl_updateService:(UInt8)service
+        allowedTransports:(nullable NSArray<SDLTransportClassBox *> *)transportList
+         primaryAvailable:(BOOL)primaryTransportAvailable
+       secondaryAvailable:(BOOL)secondaryTransportAvailable
+    transportUpdatedBlock:(void (^)(SDLProtocol * _Nullable oldProtocol, SDLProtocol * _Nullable newProtocol))transportUpdatedBlock {
+
```

Comment:
Extra line

Reply:
Fixed in commit c6d6d98423e3372a89ad1f52331bd5f1b8c65fd6.